lib/questionnaires/view/src/questionnaire_stepper_page_view.dart

```diff
@@ -0,0 +1,149 @@
+import 'package:faiadashu/questionnaires/view/item/src/questionnaire_item_filler.dart';
+import 'package:faiadashu/questionnaires/view/src/questionnaire_filler.dart';
+import 'package:faiadashu/questionnaires/view/src/questionnaire_theme.dart';
+import 'package:flutter/material.dart';
+
+/// A specialized `PageView` designed to display a series of questions
+/// in a step-by-step format.
+///
+/// [QuestionnaireStepperPageView] seamlessly integrates with [QuestionnaireStepperPageViewController]
+/// to parent widget navigating the page view, or retrieve information related the view.
+class QuestionnaireStepperPageView extends StatefulWidget {
+  final QuestionnaireStepperPageViewController? controller;
+  final ValueChanged<int>? onPageChanged;
+  final Function(bool)? onLastPageUpdated;
+  final void Function(QuestionnaireItemFiller?)? onVisibleItemUpdated;
+
+  QuestionnaireStepperPageView({
+    this.controller,
+    this.onPageChanged,
+    this.onLastPageUpdated,
+    this.onVisibleItemUpdated,
+  });
+
+  @override
+  _QuestionnaireStepperPageViewState createState() =>
+      _QuestionnaireStepperPageViewState();
+}
+
+class _QuestionnaireStepperPageViewState extends State<QuestionnaireStepperPageView> {
+  PageController _pageController = PageController();
+
+  @override
+  void initState() {
+    super.initState();
+    widget.controller?._attach(this);
+  }
+
+  /// Determines if the given index corresponds to the last page.
+  bool _hasReachedLastPage() {
+    final currentPage = _pageController.page!.round();
+    final themeData = QuestionnaireTheme.of(context);
+    final fillerData = QuestionnaireResponseFiller.of(context);
+
+    /// By checking the next item from the item builder, it verifies whether we're on the last page.
+    /// If there's no item for the next index, then we've reached the last page.
+    return themeData.stepperQuestionnaireItemFiller(
+      fillerData,
+      currentPage + 1,
+    ) == null;
+  }
+
+  /// Updates the currently visible item based on the provided index.
+  ///
+  /// This ensures that the parent context knows which item is visible and can perform any
+  /// necessary actions or updates related to that item.
+  void _updateVisibleItem(int index) {
+    final responseFiller = QuestionnaireResponseFiller.of(context);
+
+    widget.onVisibleItemUpdated?.call(responseFiller.visibleItemFillerAt(index));
```

Comment:
I think this probably should be:

```suggestion
    final data = QuestionnaireTheme.of(context)
            .stepperQuestionnaireItemFiller(responseFiller, index);
            
    widget.onVisibleItemUpdated?.call(data);
```

like we are doing in the `itemBuilder`.